XiEditor/Search.swift

```diff
@@ -0,0 +1,208 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+import Swift
+
+class FindViewController: NSViewController, NSTextFieldDelegate {
+
+    @IBOutlet weak var searchField: NSSearchField!
+    @IBOutlet weak var navigationButtons: NSSegmentedControl!
+    @IBOutlet weak var doneButton: NSButton!
+    @IBOutlet weak var ignoreCaseMenuItem: NSMenuItem!
+    @IBOutlet weak var wrapAroundMenuItem: NSMenuItem!
+
+    var optionIgnoreCase: Bool = true
+    var optionWrapAround: Bool = true
+
+    var editViewController: EditViewController!
+
+    override func viewDidLoad() {
+        // add recent searches menu items
+        let menu = searchField.searchMenuTemplate!
+
+        menu.addItem(NSMenuItem.separator())
+
+        let recentTitle = NSMenuItem(title: "Recent Searches", action: nil, keyEquivalent: "")
+        recentTitle.tag = Int(NSSearchFieldRecentsTitleMenuItemTag)
+        menu.addItem(recentTitle)
+
+        let recentItem = NSMenuItem(title: "", action: nil, keyEquivalent: "")
+        recentItem.tag = Int(NSSearchFieldRecentsMenuItemTag)
+        menu.addItem(recentItem)
+
+        menu.addItem(NSMenuItem.separator())
+
+        let recentClear = NSMenuItem(title: "Clear Recent Searches", action: nil, keyEquivalent: "")
+        recentClear.tag = Int(NSSearchFieldClearRecentsMenuItemTag)
+        menu.addItem(recentClear)
+    }
+
+    override func validateMenuItem(_ menuItem: NSMenuItem) -> Bool {
+        return true
+    }
+
+    @IBAction func selectIgnoreCaseMenuItem(_ sender: Any) {
+        if let menuItem = sender as? NSMenuItem {
+            optionIgnoreCase = !optionIgnoreCase
+            if optionIgnoreCase {
+                menuItem.state = NSOnState
+            } else {
+                menuItem.state = NSOffState
+            }
+        }
+
+        editViewController.find(searchField.stringValue,
+                                case_sensitive: !optionIgnoreCase,
+                                wrap_around: optionWrapAround)
+    }
+    
+    @IBAction func selectWrapAroundMenuItem(_ sender: Any) {
+        if let menuItem = sender as? NSMenuItem {
+            optionWrapAround = !optionWrapAround
+            if optionWrapAround {
+                menuItem.state = NSOnState
+            } else {
+                menuItem.state = NSOffState
+            }
+        }
+    }
+    
+    @IBAction func clickSegmentControl(_ sender: NSSegmentedControl) {
+        switch sender.selectedSegment {
+        case 0:
+            editViewController.findPrevious()
+        case 1:
+            editViewController.findNext()
+        default:
+            break
+        }
+    }
+
+    override func cancelOperation(_ sender: Any?) {
+        self.close()
+    }
+
+    func open() {
+        self.view.isHidden = false
+        searchField.becomeFirstResponder()
```

Comment:
`editViewController.editView.isFrontmostView = false`, and then change it back in `close()`. You will also want to set it to true if there's a mouse click in the editVIew. This isn't very elegant, just the only way I can think of at the moment. 😒